### PR TITLE
Remove holochain_keystore dep from holochain_sqlite [ TK-08526 ]

### PR DESCRIPTION
**Pull request by maackle**
We've always had this "Env" notion which is a combination of db access and keystore access, which exists for convenience. This is the only reason that holochain_sqlite needs to know about holochain_keystore. I need to let holochain_types depend on a new trait in holochain_sqlite, which creates a circular dependency. So, this PR cuts the keystore dependency.

This PR moves the "Env" concept up into holochain_state, which wraps a simpler "Db" concept in holochain_sqlite -- it just allows DB access without the KeystoreSender. Now "Env" just adds keystore access to a "Db", and there are lots of From impls to go between the two.

Most of this PR is noise. I put comments on the only two files that have anything interesting in them, and even those are not very interesting.

After merging this PR into `sqlite` branch, this will actually make the overall `sqlite -> develop` diff *less* noisy, since it will undo many of the renames from `EnvRead -> DbRead`

### CODE REVIEW

File: crates/holochain_state/src/test_utils.rs [hunk 1]
```diff
@@ -1,3 +1,148 @@
...
+
+/// A test database in a temp directory
+#[derive(Clone, Shrinkwrap)]
+pub struct TestEnv {
+    #[shrinkwrap(main_field)]
+    /// sqlite database
+    env: EnvWrite,
+    /// temp directory for this environment
+    tmpdir: Arc<TempDir>,
+}
```
Comment (maackle) [inline crates/holochain_state/src/test_utils.rs:80]: This file is one of the only ones with interesting changes

File: crates/holochain_types/src/env.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+use holochain_sqlite::prelude::*;
+use shrinkwraprs::Shrinkwrap;
+
+/// Read access to a database, plus a keystore channel sender
+#[derive(Clone, Shrinkwrap)]
+pub struct EnvRead {
+    #[shrinkwrap(main_field)]
+    db: DbRead,
+    keystore: KeystoreSender,
+}
```
Comment (maackle) [inline crates/holochain_types/src/env.rs:15]: This is the only other file of interest

**Review by neonphog — APPROVED** [review]
:+1:

**Review by freesig — APPROVED** [review]
lgtm

**Pull request merged**